OneCommunityGlobal/HighestGoodNetworkApp
# Hiral taking over for John - Weekly summary report page efficiency improvement

# Description

![Screenshot 2024-03-02 113230](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13707896/d7730edc-24f4-4201-b330-973aae337610)

## Main changes explained:
- Enhance the efficiency of the weekly summary page by improving its loading speed. 
- Implement pagination and include an option to change the number of records displayed per page.

## How to test:
1. check into the current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log in as an admin user
5. Dashboard -> Report -> Weekly Summaries Report 
6. Check for the pagination on that page. 
7. Test pagination with different numbers of records on the page. 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/66827672/afa0253f-6a3d-43e4-8a73-623ece33d553

<img width="1635" alt="Default_Pagination" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/66827672/b41e0118-0d94-47ce-9d0b-2b95cd0bb44d">

<img width="1640" alt="Items_Per_Page" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/66827672/75315d68-f50b-4913-ad18-9813b19d1b35">
